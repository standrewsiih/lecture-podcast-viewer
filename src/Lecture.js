import React from 'react';
import moment from 'moment';

const Lecture = (props) => {
  return(
    <div className="lecture">
      <div className="content-panel">    
        <h2 className="page-subheading">{props.title}</h2>
        <p className="date">{moment(props.pubDate).format('dddd MMMM Do YYYY')}</p>
        <p>{props.subtitle}</p>
        <p><audio preload="none" controls="controls"><source src={props.url} type="audio/mpeg" /></audio></p>
      </div>
    </div>
  );
};

export default Lecture;