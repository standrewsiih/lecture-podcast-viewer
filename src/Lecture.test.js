import React from 'react';
import { shallow } from 'enzyme';
import Lecture from './Lecture.js';
import moment from 'moment';

describe('Lecture', () => {

  const props = {
    title: "some lecture",
    subtitle: "some subtitle",
    summary: "some summary",
    pubDate: new Date(1487076708000),
    duration: "59:59",
    url: "https://www.intellectualhistory.net/some_lecture.mp3"
  }
  
  const lecture = shallow(<Lecture {...props}/>);
  
  it('should render properly', () => {
    expect(lecture).toMatchSnapshot();
  });

  it('should display the title', () => {
    expect(lecture.find('.page-subheading').text()).toBe('some lecture');
  });
  
  it('should display the date of publication', () => {
    expect(lecture.find('.date').text()).toBe('Tuesday February 14th 2017');
  });

  it('should contain an audio component', () => {
    expect(lecture.find('audio').exists()).toBe(true);
    expect(lecture.find('source').props().src).toBe(props.url);
  });

});