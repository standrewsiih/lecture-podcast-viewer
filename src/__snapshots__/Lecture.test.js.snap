// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lecture should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Lecture
    duration="59:59"
    pubDate={2017-02-14T12:51:48.000Z}
    subtitle="some subtitle"
    summary="some summary"
    title="some lecture"
    url="https://www.intellectualhistory.net/some_lecture.mp3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          class="page-heading"
/>,
        <div
          className="content-panel"
>
          <h2
                    className="page-subheading"
          >
                    some lecture
          </h2>
          <p
                    className="date"
          >
                    Tuesday February 14th 2017
          </p>
          <p>
                    some subtitle
          </p>
          <p>
                    <audio
                              controls="controls"
                              preload="none"
                    >
                              <source
                                        src="https://www.intellectualhistory.net/some_lecture.mp3"
                                        type="audio/mpeg"
                              />
                    </audio>
          </p>
</div>,
      ],
      "className": "lecture",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "class": "page-heading",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="page-subheading"
>
              some lecture
</h2>,
            <p
              className="date"
>
              Tuesday February 14th 2017
</p>,
            <p>
              some subtitle
</p>,
            <p>
              <audio
                            controls="controls"
                            preload="none"
              >
                            <source
                                          src="https://www.intellectualhistory.net/some_lecture.mp3"
                                          type="audio/mpeg"
                            />
              </audio>
</p>,
          ],
          "className": "content-panel",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "some lecture",
              "className": "page-subheading",
            },
            "ref": null,
            "rendered": "some lecture",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Tuesday February 14th 2017",
              "className": "date",
            },
            "ref": null,
            "rendered": "Tuesday February 14th 2017",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "some subtitle",
            },
            "ref": null,
            "rendered": "some subtitle",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <audio
                controls="controls"
                preload="none"
>
                <source
                                src="https://www.intellectualhistory.net/some_lecture.mp3"
                                type="audio/mpeg"
                />
</audio>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <source
                  src="https://www.intellectualhistory.net/some_lecture.mp3"
                  type="audio/mpeg"
/>,
                "controls": "controls",
                "preload": "none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "src": "https://www.intellectualhistory.net/some_lecture.mp3",
                  "type": "audio/mpeg",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              "type": "audio",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            class="page-heading"
/>,
          <div
            className="content-panel"
>
            <h2
                        className="page-subheading"
            >
                        some lecture
            </h2>
            <p
                        className="date"
            >
                        Tuesday February 14th 2017
            </p>
            <p>
                        some subtitle
            </p>
            <p>
                        <audio
                                    controls="controls"
                                    preload="none"
                        >
                                    <source
                                                src="https://www.intellectualhistory.net/some_lecture.mp3"
                                                type="audio/mpeg"
                                    />
                        </audio>
            </p>
</div>,
        ],
        "className": "lecture",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "class": "page-heading",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="page-subheading"
>
                some lecture
</h2>,
              <p
                className="date"
>
                Tuesday February 14th 2017
</p>,
              <p>
                some subtitle
</p>,
              <p>
                <audio
                                controls="controls"
                                preload="none"
                >
                                <source
                                                src="https://www.intellectualhistory.net/some_lecture.mp3"
                                                type="audio/mpeg"
                                />
                </audio>
</p>,
            ],
            "className": "content-panel",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "some lecture",
                "className": "page-subheading",
              },
              "ref": null,
              "rendered": "some lecture",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tuesday February 14th 2017",
                "className": "date",
              },
              "ref": null,
              "rendered": "Tuesday February 14th 2017",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "some subtitle",
              },
              "ref": null,
              "rendered": "some subtitle",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <audio
                  controls="controls"
                  preload="none"
>
                  <source
                                    src="https://www.intellectualhistory.net/some_lecture.mp3"
                                    type="audio/mpeg"
                  />
</audio>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <source
                    src="https://www.intellectualhistory.net/some_lecture.mp3"
                    type="audio/mpeg"
/>,
                  "controls": "controls",
                  "preload": "none",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "src": "https://www.intellectualhistory.net/some_lecture.mp3",
                    "type": "audio/mpeg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "source",
                },
                "type": "audio",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
